source/backend/api/Services/PropertyService.cs

```diff
@@ -192,31 +202,64 @@ public PropertyManagementModel UpdatePropertyManagement(PimsProperty property)
             return GetPropertyManagement(newProperty.Internal_Id);
         }
 
-        public IList<PimsPropPropActivity> GetManagementActivities(long propertyId)
+        public IList<PimsPropertyActivity> GetActivities(long propertyId)
         {
             _logger.LogInformation("Getting property management activities for property with id {propertyId}", propertyId);
             _user.ThrowIfNotAuthorized(Permissions.ManagementView, Permissions.PropertyView);
 
-            return _propertyRepository.GetManagementActivitiesByProperty(propertyId);
+            return _propertyActivityRepository.GetActivitiesByProperty(propertyId);
         }
 
-        public bool DeleteManagementPropPropActivity(long propertyId, long managementActivityId)
+        public PimsPropertyActivity GetActivity(long propertyId, long activityId)
         {
-            _logger.LogInformation("Deleting Management Activity with id {managementActivityId} from property with Id {propertyId}", managementActivityId, propertyId);
-            _user.ThrowIfNotAuthorized(Permissions.ManagementDelete);
+            _logger.LogInformation("Retrieving single property Activity...");
+            _user.ThrowIfNotAuthorized(Permissions.ManagementView, Permissions.PropertyView);
+
+            var propertyActivity = _propertyActivityRepository.GetActivity(activityId);
 
-            var propertyManagementActivity = _propertyRepository.GetManagementActivityById(managementActivityId);
-            if(propertyManagementActivity.PropertyId != propertyId)
+            if (propertyActivity.PimsPropPropActivities.Any(x => x.PropertyId == propertyId))
             {
-                throw new BadRequestException($"PropertyManagementActivity with Id: {managementActivityId} and PropertyId {propertyId} doesn't exist");
+                return propertyActivity;
             }
 
-            if (!propertyManagementActivity.PimsPropertyActivity.PropMgmtActivityStatusTypeCode.Equals(PropertyActivityStatusTypeCode.NOTSTARTED.ToString()))
+            throw new BadRequestException("Activity with the given id does not match the property id");
+        }
+
+        public PimsPropertyActivity CreateActivity(PimsPropertyActivity propertyActivity)
+        {
+            _logger.LogInformation("Creating property Activity...");
+            _user.ThrowIfNotAuthorized(Permissions.ManagementEdit, Permissions.PropertyEdit);
```

Comment:
same comment here, should be ManagementAdd, PropertyAdd

Reply:
how about managementadd propertyedit